android/app/src/main/java/com/msupplycoldchain/SelfSignedCertClientFactory.java

```diff
@@ -0,0 +1,121 @@
+            final SSLContext sslContext = SSLContext.getInstance("SSL");
+            sslContext.init(null, trustAllCerts, new java.security.SecureRandom());
+
+            // Create an ssl socket factory with our all-trusting manager
+            final SSLSocketFactory sslSocketFactory = sslContext.getSocketFactory();
+            OkHttpClient.Builder builder = new OkHttpClient.Builder()
+                    .connectTimeout(0, TimeUnit.MILLISECONDS).readTimeout(0, TimeUnit.MILLISECONDS)
+                    .writeTimeout(0, TimeUnit.MILLISECONDS).cookieJar(new ReactCookieJarContainer());
+            builder.sslSocketFactory(sslSocketFactory, (X509TrustManager) trustAllCerts[0]);
+            builder.addInterceptor(new Interceptor() {
+                @Override
+                public Response intercept(Chain chain) throws IOException {
+                    Request request = chain.request();
+                    // Test the host and see if this is a private IP i.e.
+                    // - 127.0.0.1
+                    // - 192.168.0.0 to 192.168.255.255
+                    // - 10.0.0.0 to 10.255.255.255
+                    // - 172.16.0.0 to 172.31.255.255
+                    Pattern pattern = Pattern.compile("(^127\\.[0-9]{1,3}\\.[0-9]{1,3}\\.[0-9]{1,3}$)|(^10\\.[0-9]{1,3}\\.[0-9]{1,3}\\.[0-9]{1,3}$)|(^172\\.1[6-9]{1}[0-9]{0,1}\\.[0-9]{1,3}\\.[0-9]{1,3}$)|(^172\\.2[0-9]{1}[0-9]{0,1}\\.[0-9]{1,3}\\.[0-9]{1,3}$)|(^172\\.3[0-1]{1}[0-9]{0,1}\\.[0-9]{1,3}\\.[0-9]{1,3}$)|(^192\\.168\\.[0-9]{1,3}\\.[0-9]{1,3}$)", Pattern.CASE_INSENSITIVE);
+                    Matcher matcher = pattern.matcher(request.url().host());
+                    isPrivate = matcher.find();
+                    // and proceed as normal with the request
+                    Response response = chain.proceed(request);
+                    return response;
+                }
+            });
+
+            builder.hostnameVerifier(new HostnameVerifier() {
+                @Override
+                public boolean verify(String hostname, SSLSession session) {
+                    return true;
+                }
+            });
+
+             OkHttpClient okHttpClient = builder.build();
+            return okHttpClient;
+        } catch (Exception e) {
+            Log.e(TAG, e.getMessage());
+            throw new RuntimeException(e);
+        }
+    }
+
+    X509TrustManager getTrustManager() throws Exception {
+        TrustManagerFactory tmf = TrustManagerFactory.getInstance("X509");
+        tmf.init((KeyStore) null);
+        TrustManager tms [] = tmf.getTrustManagers();
+
+        /*
+         * Iterate over the returned trust managers,
+         * look for an instance of X509TrustManager
+         * and return if found
+         */
+        for (TrustManager tm: tms) {
+            if (tm instanceof X509TrustManager) {
+                return (X509TrustManager) tm;
+            }
+        }
+        throw new Exception("Couldn't initialize");
```

Comment:
```suggestion
        throw new Exception("Couldn't initialize TrustManager");
```
Unlikely to happen, but I always like a more specific error (or at least searchable error) if something unexpected does happen!